src/main/java/com/duke/bot/Duke.java

```diff
@@ -0,0 +1,163 @@
+                echo();
+                break;
+
+            case "deadline":
+                ui.setToken("\\s*/by\\s*|\\n");
+                String action = ui.getNext();
+                String dateInput = ui.getNext();
+                LocalDate byDate = LocalDate.parse(dateInput);
+                DeadlineTask deadlineTask = DeadlineTask.createDeadlineTask(action, byDate);
+                tasks.addTask(deadlineTask);
+                ui.print("Added: " + deadlineTask.toString());
+                ui.print(String.format("Now you have %d task(s) on your list.\n", tasks.getSize()));
+                ui.printEmptyLine();
+                ui.setToken("\\p{javaWhitespace}+");
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            case "event":
+                ui.setToken("\\s*/at\\s*|\\n");
+                action = ui.getNext();
+                dateInput = ui.getNext();
+                LocalDate atDate = LocalDate.parse(dateInput);
+                EventTask eventTask = EventTask.createEventTask(action, atDate);
+                tasks.addTask(eventTask);
+                ui.print("Added: " + eventTask.toString());
+                ui.print(String.format("Now you have %d task(s) on your list.\n", tasks.getSize()));
+                ui.printEmptyLine();
+                ui.setToken("\\p{javaWhitespace}+");
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            case "find":
+                ui.resetScanner();
+                String keyword = ui.getNextLine();
+                for (int i = 0; i < tasks.getSize(); ++i) {
+                    Task task = tasks.getTask(i);
+                    if (task.getTaskName().contains(keyword)) {
+                        ui.print((i + 1) + ". " + task.toString());
+                    }
+                }
+                echo();
+                break;
+
+            case "bye":
+                ui.printByeMsg();
+                break;
+
+            case "delete":
+                int delIdx = ui.getNextInt() - 1;
+                if (delIdx >= tasks.getSize() || delIdx < 0) {
+                    throw new DukeException("Oops! Target object is out of bounds!");
+                }
+                Task delTask = tasks.getTask(delIdx);
+                tasks.deleteTask(delIdx);
+                ui.print(String.format("Deleted: %s\n\n", delTask.toString()));
+                storage.saveToFile(tasks.printList());
+                echo();
+                break;
+
+            default:
+                throw new DukeException(
+                        "Oops! Invalid commmand word, perhaps you would want to try on of the following: "
+                                + "1. todo 2.deadline 3.event 4.list 5.done 6.bye");
+
+            }
+
+        } catch (DukeException ex) {
+            ui.printError(ex.getMessage());
+            ui.printEmptyLine();
+            echo();
```

Comment:
Bad practice to throw and catch Exceptions in the same function.
Exceptions use more memory than normal if/else.
Use Exceptions as a 'return' statement which remind you that it might not always return the desired output.